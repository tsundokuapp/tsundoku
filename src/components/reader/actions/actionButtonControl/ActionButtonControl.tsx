interface ActionButtonControlProps {
  children: React.ReactNode;
  onClick: () => void;
  disable: boolean;
}

export const ActionButtonControl = ({
  children,
  onClick,
  disable,
  ...props
}: ActionButtonControlProps) => {
  return (
    <button
      role="button"
      onClick={onClick}
      className="inline-flex h-10 w-full items-center justify-between rounded-lg border border-zinc-800 bg-transparent px-4 py-2 text-sm font-medium text-zinc-500 hover:text-zinc-400 focus:outline-none"
      disabled={disable}
      {...props}
    >
      {children}
    </button>
  );
};
