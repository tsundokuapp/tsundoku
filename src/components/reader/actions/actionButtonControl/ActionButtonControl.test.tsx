import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';

import { ActionButtonControl } from './ActionButtonControl';

describe('Testes para o componente <ActionButtonControl />', () => {
  describe('Testes de renderização', () => {
    it('Deve renderizar corretamente com o texto esperado', () => {
      // Given
      render(
        <ActionButtonControl onClick={jest.fn()} disable={false}>
          Clique aqui
        </ActionButtonControl>,
      );

      // Then
      expect(
        screen.getByRole('button', { name: /clique aqui/i }),
      ).toBeInTheDocument();
    });
  });

  describe('Testes de interação e comportamento', () => {
    it('Deve chamar a função onClick ao ser clicado', () => {
      const handleClick = jest.fn();

      // Given
      render(
        <ActionButtonControl onClick={handleClick} disable={false}>
          Clique aqui
        </ActionButtonControl>,
      );

      // When
      fireEvent.click(screen.getByRole('button', { name: /clique aqui/i }));

      // Then
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('Deve estar desativado quando a prop disable for true', () => {
      // Given
      render(
        <ActionButtonControl onClick={jest.fn()} disable={true}>
          Clique aqui
        </ActionButtonControl>,
      );

      // When
      const button = screen.getByRole('button', { name: /clique aqui/i });

      // Then
      expect(button).toBeDisabled();
    });

    it('Não deve chamar onClick quando desativado', () => {
      const handleClick = jest.fn();

      // Given
      render(
        <ActionButtonControl onClick={handleClick} disable={true}>
          Clique aqui
        </ActionButtonControl>,
      );

      // When
      fireEvent.click(screen.getByRole('button', { name: /clique aqui/i }));

      // Then
      expect(handleClick).not.toHaveBeenCalled();
    });
  });

  describe('Teste de snapshot', () => {
    it('Deve manter o snapshot do ActionButtonControl', () => {
      // Given
      const { container } = render(
        <ActionButtonControl onClick={jest.fn()} disable={false}>
          Clique aqui
        </ActionButtonControl>,
      );

      // Then
      expect(container).toMatchSnapshot();
    });
  });
});
