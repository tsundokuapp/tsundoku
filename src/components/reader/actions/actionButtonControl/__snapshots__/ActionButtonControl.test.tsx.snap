// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes para o componente <ActionButtonControl /> Teste de snapshot Deve manter o snapshot do ActionButtonControl 1`] = `
<div>
  <button
    class="inline-flex h-10 w-full items-center justify-between rounded-lg border border-zinc-800 bg-transparent px-4 py-2 text-sm font-medium text-zinc-500 hover:text-zinc-400 focus:outline-none"
    role="button"
  >
    Clique aqui
  </button>
</div>
`;
